catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -397,7 +397,10 @@ private ContentItem readContent(URI uri) throws StorageException {
       extension = FilenameUtils.getExtension(FilenameUtils.removeExtension(filename));
       try {
         reference = new URI(new String(Files.readAllBytes(file), Charset.forName("UTF-8")));
-        if (reference.getScheme().equalsIgnoreCase("file")) {
+
+        if (reference.getScheme() == null) {
```

Comment:
❓ Is this branch necessary? What was producing a null reference scheme?

Reply:
When a monitored file was updated the URI had only the path and no scheme causing a NPE here. Add this check to prevent this. File updates seem to work fine with this change.

Reply:
These files are only written and read by the storage provider IIRC. It sounds like it is either receiving a schemeless URI or not persisting that URI correctly. I would prefer to address that issue, it sounds like a bug to me.

Reply:
Seems like we would still want at least a check here so we don't throw a NPE. It it a requirement that the Path we get back from getContentFilePath has a scheme? Would you be ok with keeping this change for now and have a new ticket look into this?

Reply:
It used to be required, hence my concern. You might try testing a filesystem monitor and a webdav monitor, I suspect something will break in that case.

Reply:
After the latest update, I tested with both a directory and a webdav endpoint both worked for ingest and update with this change.